src/libffi/ffi.c

```diff
@@ -0,0 +1,255 @@
+// Cointains all inline functions in include/notcurses/*.h
+// This file is auto generated from tools/generate_ffi.py
+#include <notcurses/notcurses.h>
+#include <notcurses/direct.h>
+#include <notcurses/nckeys.h>
+
+bool nccapability_canchangecolor(const nccapabilities* caps);
```

Comment:
this style requires us to add every new `static inline` function here explicitly, and i don't see any mechanism triggering a failure if someone forgets to. this will thus inevitably get out-of-sync over time.

oh nevermind it's being autogenerated.

still....wouldn't it be sufficient to just `include` the main headers after defining `NOTCURSES_FFI`? i.e. i think it might be possible to just have a file with

```c
#define NOTCURSES_FFI
#include <notcurses/notcurses.h>
#include <notcurses/ncdirect.h>
```

this is going to pull all the actual definitions in, and `keep-inline-functions` ought then suffice. i *think*. does that make sense?

Reply:
Yeah, I agree it makes sense. I tried something like that earlier and it didn't work immediately but I will poke at it more.

Reply:
i'm happy to do so if you want to get onto other stuff, just let me know!

Reply:
Feel free to push any improvement you have to this branch (and any future branch).

Just doing the 

```
#define NOTCURSES_FFI
#include <notcurses/notcurses.h>
#include <notcurses/ncdirect.h>
```

does not immediately work at least, not from what I can see. If I try it I start getting things like

```
In file included from /home/kc/Applications/notcurses-3.0/notcurses-3.0/include/notcurses/ncport.h:14,
                 from /home/kc/Applications/notcurses-3.0/notcurses-3.0/include/notcurses/notcurses.h:15,
                 from /home/kc/Applications/notcurses-3.0/notcurses-3.0/src/libffi/ffi.c:4:
/home/kc/Applications/notcurses-3.0/notcurses-3.0/include/notcurses/ncport.h:19:19: warning: ‘__bswap_32’ is static but used in inline function ‘ncpixel_set_b’ which is not static
   19 | #define htole(x) (__bswap_32(htonl(x)))
```

and the symbols do not end up in the resulting lirary at all. The resulting `ffi.c.o` is like 1.7k so it barely contains anyhitng. So the declarations in `ffi.c` seem to do something, at least.

Reply:
in that case, i think i'll merge this (assuming other small things have been fixed), and maybe plug away at it if so moved